sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1198,6 +1208,20 @@ case class Cast(
 
     case _ if from == NullType => (c, evPrim, evNull) => code"$evNull = true;"
     case _ if to == from => (c, evPrim, evNull) => code"$evPrim = $c;"
+    case _ if from.isInstanceOf[VariantType] => (c, evPrim, evNull) =>
+      val tmp = ctx.freshVariable("tmp", classOf[Object])
+      val dataTypeArg = ctx.addReferenceObj("dataType", to)
+      val zoneIdArg = ctx.addReferenceObj("zoneId", timeZoneId)
+      val failOnError = evalMode != EvalMode.TRY
+      code"""
+        Object $tmp = org.apache.spark.sql.catalyst.expressions.variant.VariantGet.cast(
```

Comment:
nit: use `classOf[T]` to avoid hardcoding class name

Reply:
Done.